### PR TITLE
task: add built-in environments

### PR DESCRIPTION
**Pull request by chriswk**
- Adds development and production environments.
- Connects default environment to all projects
- When creating a project connects the project to all enabled
  environments

### CODE REVIEW

File: src/migrations/20210928080601-add-development-and-production-environments.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+'use strict';
```
[the remark this answers is not in the document]
Comment (chriswk) [reply src/migrations/20210928080601-add-development-and-production-environments.js:1]: @sonatype-lift ignore

File: src/migrations/20210928080601-add-development-and-production-environments.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        `
+            INSERT INTO environments(name, type, enabled)
+            VALUES ('development', 'development', true),
+                   ('production', 'production', true);
+        `,
+        cb,
+    );
+};
+
+exports.down = function (db, cb) {
```
[the remark this answers is not in the document]
Comment (chriswk) [reply src/migrations/20210928080601-add-development-and-production-environments.js:14]: @sonatype-lift ignore

File: src/lib/db/environment-store.ts [hunk 3]
```diff
@@ -95,10 +95,14 @@ export default class EnvironmentStore implements IEnvironmentStore {
         throw new NotFoundError(`Could not find environment with name: ${key}`);
     }
 
-    async getAll(): Promise<IEnvironment[]> {
-        const rows = await this.db<IEnvironmentsTable>(TABLE)
+    async getAll(query?: Object): Promise<IEnvironment[]> {
```
Comment (ivarconr) [inline src/lib/db/environment-store.ts:98]: You are allowed to override the Object with a specific interface.

File: src/migrations/20210928080601-add-development-and-production-environments.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ivarconr) [reply src/migrations/20210928080601-add-development-and-production-environments.js:1]: @chriswk do you know if we can get lift to ignore our migrations folder all together?

**Review by ivarconr — APPROVED** [review]
lgtm.

**Pull request merged**